gis/text_to_centreline/README.md

```diff
@@ -0,0 +1,800 @@
+
+# Creating Bylaws Speed Limit Layer
+
+The function created above was to read bylaws and return the centrelines involved, be it partial or complete. The following steps then have to be done in order to prepare the final bylaws speed limit layer. 
+
+## Using the Function
+ 
+Using the function `gis.text_to_centreline`, convert all bylaws text into centrelines and put the results into a table named `gis.bylaws_routing`. The query used is as shown below and can also be found [here](https://github.com/CityofToronto/bdit_data-sources/blob/text_to_centreline/gis/text_to_centreline/sql/table-bylaws_routing.sql). **Note** that there were a couple of them that raised a warning message and you can find out more in [Quality Control](#Quality-Control) part (iii) .
+
+```sql
+SET client_min_messages = warning; 
+--only show warning messages that I would like to know
+CREATE TABLE gis.bylaws_routing AS
+SELECT law.*, results.*
+FROM jchew.bylaws_to_update law, --bylaws where deleted = false
+LATERAL gis.text_to_centreline(
+law.id,
+law.highway,
+law.between,
+NULL
+) as results
+```
+
+However, the function does not include the `date_added` and `date_repealed` which are essential in the next part. Therefore, in order to get that information, the following query has to be run too. This can be one of the [future enhancement](#Include-date_added-and-date_repealed).
+```sql
+CREATE TABLE gis.bylaws_routing_dates AS
+SELECT result.*, dates.date_added, dates.date_repealed FROM gis.bylaws_routing result
+LEFT JOIN jchew.bylaws_added_repealed_dates dates
+USING (id)
+```
+
+## Match to Centrelines and Categorize Bylaws
+
+The previous step only converts all bylaws into centrelines and do not include centrelines that are not stated in the bylaws. This [mat view query](https://github.com/CityofToronto/bdit_data-sources/blob/text_to_centreline/gis/text_to_centreline/sql/mat-view-bylaws_centreline_categorized.sql) categorizes bylaws into different parts and incorporates that into the centreline layer into a mat view named `gis.bylaws_centreline_categorized`. We check if the centrelines are involved in any bylaws, if they are not, set the speed limit to 50km/h. If they are just partially included in the bylaws, we check if there's another bylaw that governs that centreline. If there is, apply the next bylaw; If there is none, set the speed limit to 50km/h. For a centreline that is included partially in more than one bylaws, it falls into the part two category. This query may seem long but it is technically just handling the bylaws in a few parts. 
```

Comment:
The function categorizes centrelines based on how the processed bylaw geometries intersect with the original geometry (not all, entirely, partially, or multiple bylaws). Man this is challenging to communicate. I think a graphic could help of the different cases this is dealing with (if you have time 😨 )